pkg/steps/pod.go

```diff
@@ -198,3 +182,65 @@ func PodStep(name string, config PodStepConfiguration, resources api.ResourceCon
 		jobSpec:     jobSpec,
 	}
 }
+
+func (s *podStep) generatePodForStep() (*coreapi.Pod, error) {
+	pod := &coreapi.Pod{
+		ObjectMeta: meta.ObjectMeta{
+			Name: s.config.As,
+			Labels: trimLabels(map[string]string{
+				PersistsLabel:    "false",
+				JobLabel:         s.jobSpec.Job,
+				BuildIdLabel:     s.jobSpec.BuildId,
+				ProwJobIdLabel:   s.jobSpec.ProwJobID,
+				CreatedByCILabel: "true",
+			}),
+			Annotations: map[string]string{
+				JobSpecAnnotation:                     s.jobSpec.RawSpec(),
+				annotationContainersForSubTestResults: s.name,
+			},
+		},
+		Spec: coreapi.PodSpec{
+			ServiceAccountName: s.config.ServiceAccountName,
+			RestartPolicy:      coreapi.RestartPolicyNever,
+			Containers: []coreapi.Container{
+				{
+					Name:                     s.name,
+					Command:                  []string{"/bin/sh", "-c", "#!/bin/sh\nset -eu\n" + s.config.Commands},
+					TerminationMessagePolicy: coreapi.TerminationMessageFallbackToLogsOnError,
+				},
+			},
+		},
+	}
+
+	if s.config.SecretName != "" {
```

Comment:
IMHO since we will always try to get both `volume` and `volumeMount`, I think it's better to have them in one function that returns both of values, and the nil check can be done by passing the whole `Secret` struct in the config.

Reply:
Will we always have both? I thought we convenience default the mount path

Reply:
I think it just overengineering. Normally we should always have both. But for this use-case. Future might change.

Reply:
No problem from my side. That was just a nit.